bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2Client.java

```diff
@@ -114,75 +117,92 @@ public void refreshPower() {
     }
 
     public void refreshAllServices() {
-        if(power || channels.isEmpty()) {
-            transmit(PATH_ALL_SERVICES).ifPresent(this::processAllServicesResult);
+        if (power || channels.isEmpty()) {
+            transmitWithResult(PATH_ALL_SERVICES).ifPresent(this::processAllServicesResult);
         }
     }
 
     public void refreshChannel() {
-        if(power) {
-            transmit(PATH_CHANNEL).ifPresent(this::processChannelResult);
+        if (power) {
+            transmitWithResult(PATH_CHANNEL).ifPresent(this::processChannelResult);
         }
     }
 
     public void refreshAnswer() {
-        if(asking) {
-            transmit(PATH_ANSWER).ifPresent(this::processAnswerResult);
+        if (asking) {
+            transmitWithResult(PATH_ANSWER).ifPresent(this::processAnswerResult);
         }
     }
 
     public void refreshVolume() {
-        if(power) {
-            transmit(PATH_VOLUME).ifPresent(this::processVolumeResult);
+        if (power) {
+            transmitWithResult(PATH_VOLUME).ifPresent(this::processVolumeResult);
         }
     }
 
     public void refreshEpg() {
-        if(power) {
-            Optional.ofNullable(channels.get(channel)).flatMap(name -> transmit(PATH_EPG + UrlEncoded.encodeString(name))).ifPresent(this::processEpgResult);
+        if (power) {
+            Optional.ofNullable(channels.get(channel))
+                    .flatMap(name -> transmitWithResult(PATH_EPG + UrlEncoded.encodeString(name)))
+                    .ifPresent(this::processEpgResult);
         }
     }
 
-    private Optional<Document> transmit(String path) {
+    private Optional<Document> transmitWithResult(String path) {
         try {
             DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
             DocumentBuilder builder = factory.newDocumentBuilder();
-            getLogger().debug("Transmitting {}", host + path);
-            String xml = getEnigma2HttpClient().get(host + path);
-            getLogger().debug("Transmitting result is {}", xml);
-            return Optional.ofNullable(builder.parse(new InputSource(new StringReader(xml))));
+            Optional<String> xml = transmit(path);
+            if(xml.isPresent()) {
+                return Optional.ofNullable(builder.parse(new InputSource(new StringReader(xml.get()))));
+            }
+            return Optional.empty();
         } catch (IOException | SAXException | ParserConfigurationException | IllegalArgumentException e) {
             if (online || !initialized) {
-                getLogger().debug("Error on transmit {}.", host + path, e);
+                logger.debug("Error on transmit {}.", host + path, e);
+            }
+            return Optional.empty();
+        }
+    }
+
+    private Optional<String> transmit(String path) {
+        try {
+            logger.debug("Transmitting {}", host + path);
```

Comment:
You should cache `host+path` in a local variable so it can be reused.

Reply:
Fixed